core/app/models/spree/app_configuration.rb

```diff
@@ -309,6 +309,16 @@ def shipping_rate_taxer_class
       @shipping_rate_taxer_class ||= Spree::Tax::ShippingRateTaxer
     end
 
+    # Allows providing your own class for calculating taxes on a shipping rate.
+    #
+    # @!attribute [rw] shipping_rate_tax_calculator_class
+    # @return [Class] a class with the same public interfaces as
+    #   Spree::TaxCalculator::ShippingRate
+    attr_writer :shipping_rate_tax_calculator_class
```

Comment:
Oops, yes, I meant both.

The PR looks really great! I'd love for us to merge it but it would be great to have a chance to collaborate (maybe even at SolidusConf?) on the interface before committing to it.

If that sounds OK could we perhaps also mark these two new classes as `@api private` for the moment, as we had done for the `Tax::ItemAdjuster` class?